### PR TITLE
Bounty support for treasury

### PR DESCRIPTION
**Pull request by sorpaas**
See https://kusama.polkassembly.io/post/237 for how this system works.

### CODE REVIEW

**Comment by sorpaas** [comment]
From what I see there's no migration associated with the bounty system, but let me know in case I'm wrong.

File: runtime/src/lib.rs [hunk 1]
```diff
@@ -548,7 +548,13 @@ parameter_types! {
 	pub const TipCountdown: BlockNumber = 1 * DAYS;
 	pub const TipFindersFee: Percent = Percent::from_percent(20);
 	pub const TipReportDepositBase: Balance = 1 * DOLLARS;
-	pub const TipReportDepositPerByte: Balance = 10 * MILLICENTS;
+	pub const DataDepositPerByte: Balance = 1 * CENTS;
+	pub const BountyDepositBase: Balance = 1 * DOLLARS;
+	pub const BountyDepositPayoutDelay: BlockNumber = 8 * DAYS;
+	pub const BountyUpdatePeriod: BlockNumber = 90 * DAYS;
```
Comment (shawntabrizi) [inline runtime/src/lib.rs:554]: I would recommend this number be lower. This configuration basically sets how often a curator must report back an update to the network. I suggested it should be 30 days on Polkadot, but Raul wanted to be a bit more conservative.

I think 3 months per update from a curator is just too long.

**Review by shawntabrizi — APPROVED** [review]
There will be a 2.0 release of substrate potentially this monday. Would look to push this and bump vendor at that point?

File: runtime/src/lib.rs (hunk 1 above)
Comment (sorpaas) [reply runtime/src/lib.rs:554]: It's just a call to `extend_bounty_expiry` that the curator can call at any time, right? So how about we set it to lower? Something like 16 days?

Comment (shawntabrizi) [reply runtime/src/lib.rs:554]: Yeah 2-4 weeks all makes sense to me.

**Pull request merged**